cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -42,28 +42,32 @@ import (
 )
 
 const (
-	machineProviderIDIndex = "machineProviderIDIndex"
-	nodeProviderIDIndex    = "nodeProviderIDIndex"
-	defaultCAPIGroup       = "cluster.x-k8s.io"
+	machineProviderIDIndex     = "machineProviderIDIndex"
+	machinePoolProviderIDIndex = "machinePoolProviderIDIndex"
+	nodeProviderIDIndex        = "nodeProviderIDIndex"
+	defaultCAPIGroup           = "cluster.x-k8s.io"
+	machinePoolAnnotationKey   = "cluster.x-k8s.io/machinepool"
```

Comment:
This annotation actually isn't referenced by CAPI. It's here as part of a fallback strategy in case we can't look up the Node by ProviderID, but I'm guessing now that this logic doesn't work.

Let me see if it needs to be updated to reference a CAPI-owned annotation, or if this should be removed. 

Sorry for the last-minute revelation! We can extend lazy consensus if needed.

Reply:
I removed this unused annotation after consulting with @CecileRobertMichon.